[Foundry] Verification fails

Fresh create-eth instance with foundry

```
yarn verify --network sepolia
forge script script/VerifyAll.s.sol --ffi --rpc-url sepolia
[⠊] Compiling...
No files changed, compilation skipped
Error: Usage of `address(this)` detected in script contract. Script contracts are ephemeral and their addresses should not be relied upon.
Traces:
  [1799750] → new VerifyAll@0x9f7cF1d1F558E57ef88a59ac3D47214eF25B6A06
    └─ ← [Return] 8877 bytes of code

  [12487] VerifyAll::run()
    ├─ [0] VM::projectRoot() [staticcall]
    │   └─ ← [Return] "/Users/liana/Documents/prog/buidl-guidl/create-eth/qtestf/packages/foundry"
    ├─ [0] VM::toString(11155111 [1.115e7]) [staticcall]
    │   └─ ← [Return] "11155111"
    ├─ [0] VM::readFile("/Users/liana/Documents/prog/buidl-guidl/create-eth/qtestf/packages/foundry/broadcast/Deploy.s.sol/11155111/run-latest.json") [staticcall]
    │   └─ ← [Return] <file>
    └─ ← [Revert] EvmError: Revert


Error: script failed: <empty revert data>
make: *** [verify] Error 1
```

--- a/.changeset/real-garlics-cheer.md
+++ b/.changeset/real-garlics-cheer.md
@@ -1,0 +1,5 @@
+---
+"create-eth": patch
+---
+
+fix foundry verify script

--- a/templates/solidity-frameworks/foundry/packages/foundry/script/VerifyAll.s.sol
+++ b/templates/solidity-frameworks/foundry/packages/foundry/script/VerifyAll.s.sol
@@ -66,15 +66,13 @@ contract VerifyAll is Script {
         inputs[7] = vm.toString(constructorArgs);
         inputs[8] = "--watch";
 
-        try tempVm(address(vm)).tryFfi(inputs) returns (FfiResult memory f) {
-            if (f.stderr.length != 0) {
-                console.logString(string.concat("Submitting verification for contract: ", vm.toString(contractAddr)));
-                console.logString(string(f.stderr));
-            } else {
-                console.logString(string(f.stdout));
-            }
-        } catch {
-            console.logString(string.concat("Contract ", vm.toString(contractAddr), " is already verified. Skipping verification."));
+        FfiResult memory f = tempVm(address(vm)).tryFfi(inputs);
+
+        if (f.stderr.length != 0) {
+            console.logString(string.concat("Submitting verification for contract: ", vm.toString(contractAddr)));
+            console.logString(string(f.stderr));
+        } else {
+            console.logString(string(f.stdout));
         }
         return;
     }

--- a/templates/solidity-frameworks/foundry/packages/foundry/script/VerifyAll.s.sol
+++ b/templates/solidity-frameworks/foundry/packages/foundry/script/VerifyAll.s.sol
@@ -29,7 +29,7 @@ contract VerifyAll is Script {
             string.concat(root, "/broadcast/Deploy.s.sol/", vm.toString(block.chainid), "/run-latest.json");
         string memory content = vm.readFile(path);
 
-        while (this.nextTransaction(content)) {
+        while (nextTransaction(content)) {
             _verifyIfContractDeployment(content);
             currTransactionIdx++;
         }
@@ -66,19 +66,26 @@ contract VerifyAll is Script {
         inputs[7] = vm.toString(constructorArgs);
         inputs[8] = "--watch";
 
-        FfiResult memory f = tempVm(address(vm)).tryFfi(inputs);
-
-        if (f.stderr.length != 0) {
-            console.logString(string.concat("Submitting verification for contract: ", vm.toString(contractAddr)));
-            console.logString(string(f.stderr));
-        } else {
-            console.logString(string(f.stdout));
+        try tempVm(address(vm)).tryFfi(inputs) returns (FfiResult memory f) {
+            if (f.stderr.length != 0) {
+                console.logString(string.concat("Submitting verification for contract: ", vm.toString(contractAddr)));
+                console.logString(string(f.stderr));
+            } else {
+                console.logString(string(f.stdout));
+            }
+        } catch {
+            console.logString(string.concat("Contract ", vm.toString(contractAddr), " is already verified. Skipping verification."));
         }
         return;
     }
 
-    function nextTransaction(string memory content) external view returns (bool) {
-        try this.getTransactionFromRaw(content, currTransactionIdx) {
+    function nextTransaction(string memory content) internal view returns (bool) {
+        string memory hashPath = searchStr(currTransactionIdx, "hash");
+
+        try vm.parseJson(content, hashPath) returns (bytes memory hashBytes) {
+            if (hashBytes.length == 0) {
+                return false;
+            }
             return true;
         } catch {
             return false;
@@ -89,10 +96,6 @@ contract VerifyAll is Script {
         string memory root = vm.projectRoot();
         string memory path = string.concat(root, "/out/", contractName, ".sol/", contractName, ".json");
         compiledBytecode = vm.readFile(path);
-    }
-
-    function getTransactionFromRaw(string memory content, uint96 idx) external pure {
-        abi.decode(vm.parseJson(content, searchStr(idx, "hash")), (bytes32));
     }
 
     function searchStr(uint96 idx, string memory searchKey) internal pure returns (string memory) {

